python/ray/air/util/data_batch_conversion.py

```diff
@@ -48,6 +54,13 @@ def convert_batch_type_to_pandas(
         A pandas Dataframe representation of the input data.
 
     """
+    if not pd:
```

Comment:
technically we don't have to do this as missing pandas will throw ModuleNotFound error anyways, this is making it a bit more verbose similar to how we handle pyarrow in this file, let me know what you think.